apps/openmw_test_suite/misc/test_endianness.cpp

```diff
@@ -0,0 +1,122 @@
+#include <gtest/gtest.h>
+#include "components/misc/endianness.hpp"
+
+struct EndiannessTest : public ::testing::Test {};
+
+TEST_F(EndiannessTest, test_swap_endianness_inplace1)
+{
+  uint8_t zero=0x00;
+  uint8_t ff=0xFF;
+  uint8_t fortytwo=0x42;
+  uint8_t half=128;
+
+  Misc::swapEndiannessInplace(zero);
+  EXPECT_TRUE(zero == 0x00);
```

Comment:
`EXPECT_EQ(zero, 0x00)` is slightly better is such cases, because if the condition is failed, it prints the actual value of the variable.